slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala

```diff
@@ -59,63 +60,64 @@ private[slf4j] object Slf4jLoggerInternal {
     )
   }
 
-  final class Slf4jLogger[F[_]](val logger: JLogger)(implicit F: Sync[F])
-      extends SelfAwareStructuredLogger[F] {
-    override def isTraceEnabled: F[Boolean] = F.delay(logger.isTraceEnabled)
-    override def isDebugEnabled: F[Boolean] = F.delay(logger.isDebugEnabled)
-    override def isInfoEnabled: F[Boolean] = F.delay(logger.isInfoEnabled)
-    override def isWarnEnabled: F[Boolean] = F.delay(logger.isWarnEnabled)
-    override def isErrorEnabled: F[Boolean] = F.delay(logger.isErrorEnabled)
+  final class Slf4jLogger[F[_]](val logger: JLogger, sync: Sync.Type = Sync.Type.Delay)(implicit
```

Comment:
MiMa is arguing against this change https://github.com/typelevel/log4cats/actions/runs/5270003952/jobs/9948431964?pr=762#step:13:29
But since this is a package private class actually, we can workaround this warning like it's done here https://github.com/typelevel/log4cats/blob/main/core/shared/src/main/scala/org/typelevel/log4cats/PagingSelfAwareStructuredLogger.scala#L80

Reply:
Pushed some changes, let me know if it's make sense